samples/grid-pro/demo/todo-app/demo.js

```diff
@@ -0,0 +1,266 @@
+// Constants
+const categories = [
+    { value: 'CL', label: 'Cleaning' },
+    { value: 'SH', label: 'Shopping' },
+    { value: 'CK', label: 'Cooking' },
+    { value: 'GR', label: 'Gardening' },
+    { value: 'LD', label: 'Laundry' },
+    { value: 'MT', label: 'Maintenance' },
+    { value: 'OR', label: 'Organization' },
+    { value: 'PT', label: 'Pet care' },
+    { value: 'OT', label: 'Other' }
+];
+const priority = [
+    { value: 1, label: 'Low' },
+    { value: 2, label: 'Medium' },
+    { value: 3, label: 'High' }
+];
+const columns = [{
+    id: 'Done',
+    dataType: 'boolean',
+    showInForm: false,
+    className: 'centered',
+    width: 80,
+    cells: {
+        renderer: {
+            type: 'checkbox'
+        }
+    },
+    header: {
+        className: 'centered-header'
+    }
+}, {
+    id: 'Category',
+    cells: {
+        formatter: function () {
+            return categories.find(({ value }) => value === this.value)
+                ?.label || '';
+        },
+        editMode: {
+            renderer: {
+                type: 'select',
+                options: categories
+            },
+            validationRules: ['notEmpty']
+        }
+    }
+}, {
+    id: 'Task',
+    cells: {
+        editMode: {
+            validationRules: ['notEmpty']
+        }
+    }
+}, {
+    id: 'Notes'
+}, {
+    id: 'Due date',
+    dataType: 'datetime',
+    cells: {
+        formatter: function () {
+            return this.value ? new Date(this.value).toLocaleDateString(
+                navigator.language
+            ) : '';
+        },
+        editMode: {
+            renderer: {
+                type: 'dateInput'
+            }
+        }
+    }
+}, {
+    id: 'Priority',
+    className: 'centered',
+    width: 120,
+    cells: {
+        formatter: function () {
+            const map = {
+                1: '🟢',
+                2: '🟡',
+                3: '🔴'
+            };
+            return map[this.value] || '';
+        },
+        editMode: {
+            renderer: {
+                type: 'select',
+                options: priority
+            }
+        }
+    },
+    header: {
+        className: 'centered-header'
+    }
+}];
+
+// Grid setup
+Grid.grid('container', {
+    time: {
+        locale: ''
+    },
+    dataTable: {
+        columns: {
+            Done: [false, false, false, false],
+            Category: ['CL', 'SH', 'CL', 'GR'],
+            Task: [
+                'Do laundry', 'Buy groceries', 'Clean kitchen', 'Water plants '
+            ],
+            Notes: [
+                'Use gentle cycle', 'Get milk and eggs', '',
+                // eslint-disable-next-line max-len
+                'In the kitchen use 2 cups of water, in the living room use 1 cup, do not water the cactus in the garden'
```

Comment:
It's quite long, especially for the mobile.

My ideas:
- short the text
- use `<ul>` list and `<li>` for each "step".

<img width="667" alt="Screenshot 2025-06-26 at 10 47 50" src="https://github.com/user-attachments/assets/21a3f9e9-b826-42a6-ae6e-13dbbb070e06" />

2. Can we add a close button in the top right corner?
<img width="575" alt="Screenshot 2025-06-26 at 10 50 00" src="https://github.com/user-attachments/assets/a67fe49a-431f-40da-af1f-209339c04156" />

It's only a suggestion.

Reply:
It looks better, but:
1. It looks weird when editing:
![image](https://github.com/user-attachments/assets/e989b3f9-45df-4399-ae67-72f82b56c23c)
2. I'm not sure if that is a real todo app use case?